### DISCUSSION TITLE
--allowed-errors option missing

### DISCUSSION THREAD

**Discussion by martinlingstuyl** [dotnet/dev-proxy#1226] [Q&A] [answered]
### Type of issue

Code doesn't work

### Feedback

The example mentioned in the docs does not work. It appears the option `--allowed-errors` isn't available. I also don't see it when running `devproxy --help`

https://learn.microsoft.com/en-us/microsoft-cloud/dev/dev-proxy/how-to/simulate-errors-microsoft-graph-apis#configure-errors-to-simulate

### Page URL

https://learn.microsoft.com/en-us/microsoft-cloud/dev/dev-proxy/how-to/simulate-errors-microsoft-graph-apis

### Content source URL

https://github.com/MicrosoftDocs/microsoft-cloud/blob/main/docs/dev/dev-proxy/how-to/simulate-errors-microsoft-graph-apis.md

### Author

@waldekmastykarz

### Document Id

cabe424a-1dbb-fd37-3752-4bc706f1016a

**Comment by waldekmastykarz** [comment]
Can you share the contents of your devproxyrc.json file?

**Comment by martinlingstuyl** [comment]
Hi @waldekmastykarz, here it is, but does that matter for a missing option?

```json
{
  "$schema": "https://raw.githubusercontent.com/microsoft/dev-proxy/main/schemas/v0.17.1/rc.schema.json",
  "plugins": [
    {
      "name": "RetryAfterPlugin",
      "enabled": true,
      "pluginPath": "~appFolder/plugins/dev-proxy-plugins.dll"
    },
    {
      "name": "GenericRandomErrorPlugin",
      "enabled": true,
      "pluginPath": "~appFolder/plugins/dev-proxy-plugins.dll",
      "configSection": "genericRandomErrorPlugin"
    }
  ],
  "urlsToWatch": [
    "https://graph.microsoft.com/*",
    "https://jsonplaceholder.typicode.com/*"
  ],
  "genericRandomErrorPlugin": {
    "errorsFile": "devproxy-errors.json"
  },
  "rate": 50,
  "labelMode": "text",
  "logLevel": "information",
  "newVersionNotification": "stable"
}
```

**Comment by waldekmastykarz** [comment, marked as answer]
You're using the `GenericRandomErrorPlugin`. `--allowed-errors` is specific to the `GraphRandomErrorPlugin`: https://learn.microsoft.com/en-us/microsoft-cloud/dev/dev-proxy/technical-reference/graphrandomerrorplugin

**Comment by waldekmastykarz** [comment]
> Hi @waldekmastykarz, here it is, but does that matter for a missing option?

A while back, we refactored options, so that plugin-specific options are only exposed, when the corresponding plugin is enabled. Otherwise, you'd get an option that doesn't affect how Dev Proxy work which would be confusing.

**Comment by martinlingstuyl** [comment]
💡Ah, I see now! Sorry, I've just completely misread this. It turns out there is no issue at all, except between the chair and the screen over here 😅 Let's close this ASAP!

**Comment by waldekmastykarz** [comment]
All good! Better safe than sorry. Thanks for reaching out
